visualization/include/visualization/element.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <vector>
+#include <utility>
+#include <string>
+
+namespace visualization::builder {
+
+    class Element {
+    public:
+        friend class ElementBuilder;
+    
+        const std::vector<std::pair<std::string, std::string>> get_attributes() const;
+        const std::string get_tag() const;
+
+    private: 
+        Element() = default;
+        std::string tag;
+        std::vector<std::pair<std::string, std::string>> attributes;
+    };
+
+    class ElementBuilder {
+    public: 
+        ElementBuilder& add_tag(const std::string name);
+        ElementBuilder& add_attribute(const std::string name, const std::string value);
```

Comment:
Generally, with a builder pattern, a fluent API like naming is more appropriate. `with_attribute` makes more sense when you are chaining calls.